src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -825,27 +824,91 @@ public void BlockDataDownloadedIsCalled_ForValidBlocksAfterFv_ResultShouldBeTrue
+        /// Call PartialOrFullValidationFailed and after make sure that there is a single US_CONSTANT on CT.
+        /// </summary>
+        [Fact]
+        public void ConsensusAndHeadersAreAtBlock5_Block6Presented_PartialOrFullValidationFailedCalled_ThereShouldBeASingleConstant()
+        {
+            // Chain header tree setup. Initial chain has 5 headers.
+            // Example: h1=h2=h3=h4=h5.
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder()
+                .WithInitialChain(initialChainSize)
+                .Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            // Extend a chain by 1 header.
+            // Example: h1=h2=h3=h4=h5=h6.
+            chainTip = ctx.ExtendAChain(1, chainTip);
+            List<BlockHeader> listOfCurrentChainHeaders = ctx.ChainedHeaderToList(chainTip, 1);
+
+            // Present header by peer with id 1 and then call PartialOrFullValidationFailed on it.
+            const int peerId = 1;
+            cht.ConnectNewHeaders(peerId, listOfCurrentChainHeaders);
+            List<int> validationResult = cht.PartialOrFullValidationFailed(chainTip);
+
+            // Validation result should contain 1 result pointing to CT.
+            validationResult.Should().HaveCount(1);
+            validationResult.First().Should().Be(peerId);
+        }
+
+        /// <summary>
+        /// Issue 26 @ CT is at block 5. Call PartialValidationSucceeded with header 6. ConsensusTipChanged on block 6.
+        /// Make sure US_CONSTANT moved to 6.
+        /// </summary>
+        [Fact]
+        public void ConsensusAndHeadersAreAtBlock5_Block6Presented_PartialValidationSucceededCalled_LocalPeerIdIsMovedTo6()
+        {
+            // Chain header tree setup. Initial chain has 5 headers.
+            // Example: h1=h2=h3=h4=h5.
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder()
+                .WithInitialChain(initialChainSize)
+                .Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            // Extend a chain by 1 header.
+            // Example: h1=h2=h3=h4=h5=h6.
+            chainTip = ctx.ExtendAChain(1, chainTip);
+            List<BlockHeader> listOfCurrentChainHeaders = ctx.ChainedHeaderToList(chainTip, 1);
+
+            // Present header by peer with id 1 and then call PartialValidationSucceeded on it.
+            const int peerId = 1;
+            cht.ConnectNewHeaders(peerId, listOfCurrentChainHeaders);
+            cht.PartialValidationSucceeded(chainTip, out bool reorgRequired);
+            reorgRequired.Should().BeTrue();
+
+            // Call ConsensusTipChanged on chaintip at header 6.
+            cht.ConsensusTipChanged(chainTip);
+            cht.GetPeerTipsByPeerId().Should().ContainKey(1);
```

Comment:
Similarly as above, we need to check which claims are present on the five and which claims are present on header six. The local PID is supposed to move from 5 to 6. So the end it should not be five anymore it should be on six.

Reply:
see peerIdsByTipHash